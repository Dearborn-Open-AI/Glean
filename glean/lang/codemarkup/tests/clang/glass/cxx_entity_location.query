query: | 
  codemarkup.FileEntityLocations { entity = { cxx = E } }; 
  codemarkup.cxx.CxxEntityLocation { entity = E }; 
perf: true
transform: [gensort, []]
