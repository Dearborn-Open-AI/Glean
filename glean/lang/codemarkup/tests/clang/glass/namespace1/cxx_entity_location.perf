{
  "@generated": null,
  "cxx1.DeclarationNameSpan.5": 32,
  "cxx1.FunctionDefinition.5": 3,
  "cxx1.NamespaceDefinition.5": 2,
  "cxx1.RecordDefinition.5": 2,
  "cxx1.Trace.5": 4
}