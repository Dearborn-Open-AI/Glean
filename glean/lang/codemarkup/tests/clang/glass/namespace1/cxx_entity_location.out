[
  "@generated",
  {
    "key": {
      "entity": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "f" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "foo" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 16,
                "columnBegin": 1,
                "lineEnd": 16,
                "columnEnd": 8
              }
            }
          }
        }
      },
      "location": {
        "name": "f",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 16,
            "columnBegin": 1,
            "lineEnd": 16,
            "columnEnd": 8
          }
        },
        "span": { "start": 284, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "foo" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 1,
                "lineEnd": 25,
                "columnEnd": 10
              }
            }
          }
        }
      },
      "location": {
        "name": "foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 25,
            "columnBegin": 1,
            "lineEnd": 25,
            "columnEnd": 10
          }
        },
        "span": { "start": 415, "length": 3 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "g" } } },
                  "scope": {
                    "namespace_": {
                      "key": {
                        "name": { "key": "bar" },
                        "parent": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 30,
                "columnBegin": 1,
                "lineEnd": 30,
                "columnEnd": 8
              }
            }
          }
        }
      },
      "location": {
        "name": "g",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 30,
            "columnBegin": 1,
            "lineEnd": 30,
            "columnEnd": 8
          }
        },
        "span": { "start": 458, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "h" } } },
                  "scope": { "global_": { } }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "_i" }, "type": { "key": "int" } }
                  ]
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 41,
                "columnBegin": 1,
                "lineEnd": 41,
                "columnEnd": 14
              }
            }
          }
        }
      },
      "location": {
        "name": "h",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 41,
            "columnBegin": 1,
            "lineEnd": 41,
            "columnEnd": 14
          }
        },
        "span": { "start": 538, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "_i" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": { "key": { "name": { "key": "h" } } },
                        "scope": { "global_": { } }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 2, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 41,
                "columnBegin": 8,
                "lineEnd": 41,
                "columnEnd": 13
              }
            }
          }
        }
      },
      "location": {
        "name": "_i",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 41,
            "columnBegin": 8,
            "lineEnd": 41,
            "columnEnd": 13
          }
        },
        "span": { "start": 544, "length": 2 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "fooI" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": { "key": { "name": { "key": "fooFn" } } },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "foo" } } }
                        }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 2, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 34,
                "columnBegin": 11,
                "lineEnd": 34,
                "columnEnd": 18
              }
            }
          }
        }
      },
      "location": {
        "name": "fooI",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 34,
            "columnBegin": 11,
            "lineEnd": 34,
            "columnEnd": 18
          }
        },
        "span": { "start": 481, "length": 4 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "i" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": { "key": { "name": { "key": "h" } } },
                        "scope": { "global_": { } }
                      }
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "local": { "kind": 2, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 43,
                "columnBegin": 8,
                "lineEnd": 43,
                "columnEnd": 12
              }
            }
          }
        }
      },
      "location": {
        "name": "i",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 43,
            "columnBegin": 8,
            "lineEnd": 43,
            "columnEnd": 12
          }
        },
        "span": { "start": 561, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "s" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": { "key": { "name": { "key": "h" } } },
                        "scope": { "global_": { } }
                      }
                    }
                  }
                }
              },
              "type": { "key": "foo::S" },
              "kind": { "local": { "kind": 0, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 45,
                "columnBegin": 3,
                "lineEnd": 45,
                "columnEnd": 10
              }
            }
          }
        }
      },
      "location": {
        "name": "s",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 45,
            "columnBegin": 3,
            "lineEnd": 45,
            "columnEnd": 10
          }
        },
        "span": { "start": 585, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "t" },
                  "scope": {
                    "local": {
                      "key": {
                        "name": { "key": { "name": { "key": "h" } } },
                        "scope": { "global_": { } }
                      }
                    }
                  }
                }
              },
              "type": { "key": "foo::bar::T" },
              "kind": { "local": { "kind": 0, "attribute": 0 } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 46,
                "columnBegin": 3,
                "lineEnd": 46,
                "columnEnd": 15
              }
            }
          }
        }
      },
      "location": {
        "name": "t",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 46,
            "columnBegin": 3,
            "lineEnd": 46,
            "columnEnd": 15
          }
        },
        "span": { "start": 602, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "function_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": { "name": { "key": "foo" } } },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "signature": {
                    "key": { "returns": { "key": "void" }, "parameters": [ ] }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 19,
                    "columnBegin": 1,
                    "lineEnd": 21,
                    "columnEnd": 1
                  }
                }
              },
              "isInline": false
            }
          }
        }
      },
      "location": {
        "name": "foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 19,
            "columnBegin": 1,
            "lineEnd": 21,
            "columnEnd": 1
          }
        },
        "span": { "start": 318, "length": 3 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "function_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": { "name": { "key": "fooFn" } } },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "signature": {
                    "key": {
                      "returns": { "key": "int" },
                      "parameters": [
                        { "name": { "key": "fooI" }, "type": { "key": "int" } }
                      ]
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 34,
                    "columnBegin": 1,
                    "lineEnd": 36,
                    "columnEnd": 1
                  }
                }
              },
              "isInline": false
            }
          }
        }
      },
      "location": {
        "name": "fooFn",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 34,
            "columnBegin": 1,
            "lineEnd": 36,
            "columnEnd": 1
          }
        },
        "span": { "start": 471, "length": 5 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "function_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": { "name": { "key": "h" } } },
                      "scope": { "global_": { } }
                    }
                  },
                  "signature": {
                    "key": {
                      "returns": { "key": "void" },
                      "parameters": [
                        { "name": { "key": "i" }, "type": { "key": "int" } }
                      ]
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 43,
                    "columnBegin": 1,
                    "lineEnd": 51,
                    "columnEnd": 1
                  }
                }
              },
              "isInline": false
            }
          }
        }
      },
      "location": {
        "name": "h",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 43,
            "columnBegin": 1,
            "lineEnd": 51,
            "columnEnd": 1
          }
        },
        "span": { "start": 555, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "namespace_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "bar" },
                      "parent": { "key": { "name": { "key": "foo" } } }
                    }
                  },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 27,
                    "columnBegin": 1,
                    "lineEnd": 32,
                    "columnEnd": 1
                  }
                }
              },
              "members": { "key": [ ] }
            }
          }
        }
      },
      "location": {
        "name": "bar",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 27,
            "columnBegin": 1,
            "lineEnd": 32,
            "columnEnd": 1
          }
        },
        "span": { "start": 433, "length": 3 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "namespace_": {
            "key": {
              "declaration": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 9,
                    "columnBegin": 1,
                    "lineEnd": 38,
                    "columnEnd": 1
                  }
                }
              },
              "members": { "key": [ ] }
            }
          }
        }
      },
      "location": {
        "name": "foo",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 9,
            "columnBegin": 1,
            "lineEnd": 38,
            "columnEnd": 1
          }
        },
        "span": { "start": 231, "length": 3 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "S" },
                      "scope": {
                        "namespace_": { "key": { "name": { "key": "foo" } } }
                      }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 11,
                    "columnBegin": 1,
                    "lineEnd": 11,
                    "columnEnd": 11
                  }
                }
              },
              "bases": [ ],
              "members": { "key": [ ] }
            }
          }
        }
      },
      "location": {
        "name": "S",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 11,
            "columnBegin": 1,
            "lineEnd": 11,
            "columnEnd": 11
          }
        },
        "span": { "start": 245, "length": 1 }
      }
    }
  },
  {
    "key": {
      "entity": {
        "defn": {
          "record_": {
            "key": {
              "declaration": {
                "key": {
                  "name": {
                    "key": {
                      "name": { "key": "T" },
                      "scope": {
                        "namespace_": {
                          "key": {
                            "name": { "key": "bar" },
                            "parent": { "key": { "name": { "key": "foo" } } }
                          }
                        }
                      }
                    }
                  },
                  "kind": { "struct_": { } },
                  "source": {
                    "file": { "key": "test.cpp" },
                    "lineBegin": 29,
                    "columnBegin": 1,
                    "lineEnd": 29,
                    "columnEnd": 11
                  }
                }
              },
              "bases": [ ],
              "members": { "key": [ ] }
            }
          }
        }
      },
      "location": {
        "name": "T",
        "file": { "key": "test.cpp" },
        "location": {
          "range": {
            "file": { "key": "test.cpp" },
            "lineBegin": 29,
            "columnBegin": 1,
            "lineEnd": 29,
            "columnEnd": 11
          }
        },
        "span": { "start": 447, "length": 1 }
      }
    }
  }
]